include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,398 @@
+			const auto currentChannelCount = asset::getFormatChannelCount(inFormat);
+			const auto arrayLayers = state->inImage->getCreationParameters().arrayLayers;
+			static constexpr auto maxChannels = 4u;
+
+			uint8_t* const totalImageCacheScratch = state->imageTotalCacheScratch = state->scratchMemory + state->cachesOffset;
+			uint8_t* const totalImageCacheOutput = state->imageTotalCacheOutput = state->imageTotalCacheScratch + state->imageTotalScratchCacheByteSize;
+
+			#ifdef _IRR_DEBUG
+			memset(scratchMemory, 0, state->scratchMemoryByteSize);
+			#endif // _IRR_DEBUG
+
+			const auto&& [copyInBaseLayer, copyOutBaseLayer, copyLayerCount] = std::make_tuple(state->inBaseLayer, state->outBaseLayer, state->layerCount);
+
+			auto resetState = [&]()
+			{
+				state->inBaseLayer = copyInBaseLayer;
+				state->outBaseLayer = copyOutBaseLayer;
+				state->layerCount = copyLayerCount;
+			};
+
+			core::vector3du32_SIMD localCoord;
+			for (auto& w = localCoord[3] = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
+			{
+				std::array<decodeType, maxChannels> minDecodeValues = {};
+				std::array<decodeType, maxChannels> maxDecodeValues = {};
+
+				state->inBaseLayer += w;
+				state->outBaseLayer += w;
+				state->layerCount = 1u;
+
+				auto perOutputRegionDecode = [&](const CommonExecuteData& commonExecuteData, CBasicImageFilterCommon::clip_region_functor_t& clip) -> bool
+				{
+					/*
+						Since regions may specify areas in a layer on a certain mipmap - it is desired to
+						decode an entire image itereting through all the regions overlapping as first
+					*/
+
+					const auto blockDims = asset::getBlockDimensions(commonExecuteData.inFormat);
+					const auto strides = commonExecuteData.oit->getByteStrides(TexelBlockInfo(commonExecuteData.inFormat)); // ITS WRONG
+
+					/*
+						if I were to use it I should have access to current handled region bellow
+						OIT isn't valid for decode to use strides, it will fail
+					*/
+
+					auto decode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
+					{
+						std::array<decodeType, maxChannels> decodeBuffer = {};
+						
+						/*
+							OFFSET is < 0, it will fail
+						*/
+
+						auto localOutPos = core::vectorSIMDi32(readBlockPos * blockDims + commonExecuteData.offsetDifference); // @devshgraphicsprogramming @Criss When it's < 0 it gets weird values so it crashes
+						
+						/*
+							need to get access the region somehow (in region, not out)
+						*/
+
+						auto* inDataAdress = commonExecuteData.inData + commonExecuteData.oit->getByteOffset(localOutPos, strides); //  @devshgraphicsprogramming @Criss there should be something like IN strides in commonData 
+
+						//auto* inDataAdress = commonExecuteData.inData + readBlockArrayOffset;
+						decodeType* outDataAdress = scratchMemory + ((readBlockPos.z * state->extent.height + readBlockPos.y) * state->extent.width + readBlockPos.x) * currentChannelCount;
```

Comment:
thats wrong for sure, `readBlockPos` is an absolute position in input image. What you want is position in processed sub-image (with extent defined by `state->extent`)

Reply:
did you read my code and verify that `readBlockPos` is relateive to mip-map's origin @Crisspl ?

Reply:
yes, it's absolute position within mip-level

Reply:
But really, Matt, this filter API needs docs, lots of it. I've been struggling with it as well while developing padded copy filter. And now while helping Arek because i forgot some things

Reply:
it should have been documented

Reply:
> But really, Matt, this filter API needs docs, lots of it. I've been struggling with it as well while developing padded copy filter. And now while helping Arek because i forgot some things

> it should have been documented

My suggestion is that, everytime you find something "non obvious" document it.

Reply:
the request for documentation has been pushed to our discord, TODO!